magma_access_gateway_installer/agw_network_configurator.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import os
+from subprocess import check_call, check_output
+
+import ipcalc  # type: ignore[import]
+import yaml
+from systemd.journal import JournalHandler  # type: ignore[import]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(JournalHandler())
+logger.setLevel(logging.INFO)
+
+
+class AGWInstallerNetworkConfigurator:
+
+    DNS_ADDRESSES = "8.8.8.8 208.67.222.222"
+    INTERFACES_DIR = "/etc/network/interfaces.d"
+    REBOOT_NEEDED = False
+
+    def __init__(self, network_interfaces, eth0_address, eth0_gw_ip_address):
```

Comment:
Add typing for `network_interfaces` `eth0_address`, `eth0_gw_ip_address`

Reply:
Done.